bob/src/hw_metrics_collector.rs

```diff
@@ -235,54 +321,82 @@ impl DiskStatCollector {
             let path_str = path.as_os_str().to_str().unwrap();
             if let Ok(dev_name) = Self::dev_name(path_str) {
                 let metric_prefix = format!("{}.{}", HW_DISKS_FOLDER, disk_name);
-                disk_metric_data.insert(dev_name, metric_prefix);
+                disk_metric_data.insert(dev_name, DiskStatsContainer::new(metric_prefix));
             } else {
                 warn!("Device name of disk {} is unknown", disk_name);
             }
         }
 
         DiskStatCollector {
-            iostat_avl: true,
+            procfs_avl: true,
             disk_metric_data,
+            upd_timestamp: Instant::now(),
         }
     }
 
-    fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
-        let iostat_lines = self.collect_iostat()?;
-        for i in iostat_lines {
-            let lsp: Vec<&str> = i.split_whitespace().collect();
-            if lsp.len() == 21 {
-                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[0]) {
-                    if let (Ok(rs), Ok(ws)) = (
-                        // readops per s count is in 1st column
-                        Self::parse_iostat_result(lsp[1]),
-                        // writeops per s count is in 7th column
-                        Self::parse_iostat_result(lsp[7]),
-                    ) {
-                        let iops = rs + ws;
-
-                        let gauge_name = format!("{}_iops", metric_prefix);
-                        gauge!(gauge_name, iops);
-                    } else {
-                        return Err(CommandError::Primary("Can't parse iostat output into float".to_string()));
-                    }
+    fn parse_stat_line(parts: Vec<&str>) -> Result<DiskStats, CommandError> {
+        let mut new_ds = DiskStats {
+            reads: 0,
+            writes: 0,
+            io_time: 0,
+            extended: parts.len() >= 13,
+        };
+        if let (Ok(r_ios), Ok(w_ios)) = (
+            // successfull reads count is in 3rd column
+            parts[3].parse::<u64>(),
+            // successfull writes count is in 7th column
+            parts[7].parse::<u64>(),
+        ) {
+            new_ds.reads = r_ios;
+            new_ds.writes = w_ios;  
+            if new_ds.extended {
+                // time spend doing i/o operations is in 13th column
+                if let Ok(io_time) = parts[13].parse::<u64>() {
```

Comment:
Indices are zero based, so column 13 is at index 12
```suggestion
                if let Ok(io_time) = parts[12].parse::<u64>() {
```